pl_bolts/losses/giou.py

```diff
@@ -0,0 +1,38 @@
+"""
+Generalized Intersection over Union (GIoU) loss (Rezatofighi et. al)
+"""
+
+import torch
+
+
+def giou_loss(pred: torch.Tensor, target: torch.Tensor) -> torch.Tensor:
+    """
+    Calculates the generalized intersection over union loss.
+
+    Args:
+        pred: batch of prediction bounding boxes with representation
+              [x_min, y_min, x_max, y_max]
+        target: batch of target bounding boxes with representation
+                [x_min, y_min, x_max, y_max]
+
+    Returns:
+        GIoU loss
+    """
+    eps = 1e-6
+    x_min = torch.max(pred[:, 0], target[:, 0])
+    y_min = torch.max(pred[:, 1], target[:, 1])
+    x_max = torch.min(pred[:, 2], target[:, 2])
+    y_max = torch.min(pred[:, 3], target[:, 3])
```

Comment:
`min` and `max` seem swapped?

Reply:
This is actually intended behavior! `x_min` refers to the minimum x value for the intersection between the prediction bbox and the target bbox, and you derive that value by getting the max between the `x_min` of the prediction bbox and the `x_min` of the target bbox.

Reply:
I'd refer you to [Generalized Intersection over Union](https://giou.stanford.edu/), which contains super useful visualizations and explanations for how it works 🙂

Reply:
I see, then let's keep it as is!